homeassistant/loader.py

```diff
@@ -294,32 +288,17 @@ def resolve_from_root(
                 _LOGGER.error(
                     "Error parsing manifest.json file at %s: %s", manifest_path, err
                 )
-                continue
 
-            return cls(
-                hass, f"{root_module.__name__}.{domain}", manifest_path.parent, manifest
-            )
+            else:
```

Comment:
This change doesn't look needed?

Reply:
Separation is needed, I can move this to inside the `try` section, but it's cleaner to use `else` right?

Reply:
We had a `continue` in the exception block so the old code seems to do the same as this code.

Reply:
My bad... reverting this